DOC: typo in the docstring of `random.multinomial`

ev-br opened the pull request:
The last example in the docstring of np.random.multinomial is a bit confusing: it talks about a "loaded dice" but draws variates from a five-element distribution. 

It seems to me that the example was supposed to demonstrate the convention that the last element of the `pvals` array is modified to make the probabilities sum to one:

```
In [26]: rndm = np.random.RandomState(1234)

In [27]: rndm.multinomial(1000, [1/7.]*6)
Out[27]: array([137, 142, 148, 136, 164, 273])

In [28]: rndm = np.random.RandomState(1234)

In [29]: rndm.multinomial(1000, [1/7.]*5 + [2/7.])
Out[29]: array([137, 142, 148, 136, 164, 273])
```

Hence modify the docstring to match the code. 

[ci skip]

ev-br committed:
DOC: typo in the docstring of random.multinomial

Discuss a loaded dice with six sides. Also add the
text about handling of input probabilities, as written
by Robert Kern in gh-6612.

[ci skip]

charris commented on numpy/random/mtrand/mtrand.pyx:
```diff
@@ -4460,8 +4460,8 @@ cdef class RandomState:
 
-        >>> np.random.multinomial(100, [1/7.]*5)
-        array([13, 16, 13, 16, 42])
+        >>> np.random.multinomial(100, [1/7.]*6)
```
Seems a bit obscure, why not `np.random.multinomial(100, [1/7.]*5 + [2/7.])` ? Also, the examples are not deterministic unless the random seed is set.

ev-br replied:
Yes, examples are not deterministic (are they anywhere in numpy?). This is an obscure feature, I agree --- and I suspect the docstrings was like this by design, specifically to illustrate the convention. Also, the convention *should* be illustrated I think, esp given that eg R decides instead to scale the probabilities by dividing by the sum.
In  https://github.com/scipy/scipy/pull/5410 I wrote it a bit more explicitly,https://github.com/scipy/scipy/pull/5410/files#diff-3f67e7fdb1ce6a44c0b49df2da9889c5R2290. Can do it here as well if you want me to.

charris replied:
>Yes, examples are not deterministic (are they anywhere in numpy?)

I was musing that if we ever start testing docstrings there is going to be a problem.

charris replied:
Yes, a bit more explanation of the example would help. I confess to liking the R convention better ;)

rkern replied:
The examples should primarily show you how to use it properly rather than all of the legal but confusing ways to abuse the API. The convention is intended to avoid floating point problems under the assumption that you are giving it values that are already normalized to the best of your ability within floating point `eps`. Under no circumstances should you ever intentionally give it grossly unnormalized values.

rkern replied:
A possible example warning against assuming the other convention:

```
The probability inputs should already be normalized. The value of the last entry is always
ignored and assumed to take up any leftover probability mass. To sample a biased coin
which has twice as much weight on one side than the other should *not* be sampled like so:

>>> np.random.multinomial(100, [1.0, 2.0])  # WRONG
array([100,   0])

but rather, like so:

>>> np.random.multinomial(100, [1.0 / 3, 2.0 / 3])  # RIGHT
array([38, 62])
```

ev-br commented:
I've replaced the example by a less obscure full list of probabilities, and added Robert's text verbatim. Commit authorship should likely be transferred to @rkern.

charris commented:
@ev-br Thanks. I've rewritten the comment cribbed from rkern in #6858, so closing this. You can continue the discussion on the new PR is you wish.
